src/authentication/FlowDetails.tsx

```diff
@@ -0,0 +1,240 @@
+import {
+  DataList,
+  Label,
+  PageSection,
+  Toolbar,
+  ToolbarContent,
+  ToggleGroup,
+  ToggleGroupItem,
+  AlertVariant,
+} from "@patternfly/react-core";
+import { CheckCircleIcon, TableIcon } from "@patternfly/react-icons";
+
+import type AuthenticationExecutionInfoRepresentation from "keycloak-admin/lib/defs/authenticationExecutionInfoRepresentation";
+import type AuthenticationFlowRepresentation from "keycloak-admin/lib/defs/authenticationFlowRepresentation";
+import type { FlowParams } from "./routes/Flow";
+import { ViewHeader } from "../components/view-header/ViewHeader";
+import { useAdminClient, useFetch } from "../context/auth/AdminClient";
+import { EmptyExecutionState } from "./EmptyExecutionState";
+import { toUpperCase } from "../util";
+import { FlowHeader } from "./components/FlowHeader";
+import { FlowRow } from "./components/FlowRow";
+import { ExecutionList, IndexChange, LevelChange } from "./execution-model";
+import { FlowDiagram } from "./components/FlowDiagram";
+import { useAlerts } from "../components/alert/Alerts";
+
+export type ExpandableExecution = AuthenticationExecutionInfoRepresentation & {
+  executionList: ExpandableExecution[];
+  isCollapsed: boolean;
+};
+
+export const FlowDetails = () => {
+  const { t } = useTranslation("authentication");
+  const adminClient = useAdminClient();
+  const { addAlert } = useAlerts();
+  const { id, usedBy, builtIn } = useParams<FlowParams>();
+  const [key, setKey] = useState(0);
+  const refresh = () => setKey(new Date().getTime());
+
+  const [tableView, setTableView] = useState(true);
+  const [flow, setFlow] = useState<AuthenticationFlowRepresentation>();
+  const [executionList, setExecutionList] = useState<ExecutionList>();
+  const [dragged, setDragged] =
+    useState<AuthenticationExecutionInfoRepresentation>();
+  const [liveText, setLiveText] = useState("");
+
+  useFetch(
+    async () => {
+      const flows = await adminClient.authenticationManagement.getFlows();
+      const flow = flows.find((f) => f.id === id);
+      const executions =
+        await adminClient.authenticationManagement.getExecutions({
+          flow: flow?.alias!,
+        });
+      return { flow, executions };
+    },
+    ({ flow, executions }) => {
+      setFlow(flow);
+      setExecutionList(new ExecutionList(executions));
+    },
+    [key]
+  );
+
+  const executeChange = async (
+    ex: AuthenticationFlowRepresentation,
+    change: LevelChange | IndexChange
+  ) => {
+    try {
+      let id = ex.id!;
+      if ("parent" in change) {
+        await adminClient.authenticationManagement.delExecution({ id });
+        const result =
+          await adminClient.authenticationManagement.addExecutionToFlow({
+            flow: change.parent?.displayName! || flow?.alias!,
+            provider: ex.providerId!,
+          });
+        id = result.id!;
+      }
+      const c = change as IndexChange;
+      const times = c.newIndex - c.oldIndex;
```

Comment:
If I remove this cast the code still works, so I think it's not needed.

```suggestion
      const times = change.newIndex - change.oldIndex;
```

Reply:
yeah that is a part of old code I think `LevelChange` has the same properties as `IndexChange` now